planner/property/physical_property.go

```diff
@@ -161,6 +161,15 @@ func GetCollateNameByIDForPartition(collateID int32) string {
 	return collate.CollationID2Name(collateID)
 }
 
+// cteConsumerStatus indicates whether we can let the current CTE consumer/reader be executed on the MPP nodes.
+type cteConsumerStatus int
+
+const (
+	NoCTE cteConsumerStatus = iota
+	SomeCTEFailedMpp
+	AllCTECanMpp
+)
```

Comment:
I think
1. the name should be "producer" status instead of "consumer" status.
2. two values are enough.